### ISSUE TITLE
nullable field in document query should not be optional

### ISSUE THREAD

**Issue by frantic1048** [dotansimha/graphql-code-generator#699]
**Describe the bug**
When generating a query in document contains nullable field, the generated document query's field's type is still `T | null | undefined`, which should be `T | null`

**To Reproduce**

*schema*:

```gql
# schema.graphql

type Person {
  name: String!
  age: Int
}

type Query {
  person: Person
}
```

*document*:

```gql
# document.graphql

query search {
  person {
    name
    age
  }
}
```

*command*:

```sh
gql-gen --template graphql-codegen-typescript-template --schema schema.graphql document.graphql --out schema.ts
```

**Expected behavior**

In generated `schema.ts`:

```gql
// ...

export interface Person {
  name: string;
  age?: number | null; // <- this is correct because it's introspection
}

// ...

export namespace Search {
  export type Variables = {};

  export type Query = {
    __typename?: "Query";
    person?: Person | null;
  };

  export type Person = {
    __typename?: "Person";
    name: string;
    age?: number | null; // <- this should not be optional because we queried this field
  };
}
```

`Search.Person.age` should not be optional(undefined), because we queried that nullable field, it could only be null when there is no data available.

**Environment:**

- OS: Arch Linux (latest)
- Codegen: 0.12.6
-  Node: 10.11.0
- graphql: 14.0.2

**Comment by moonray** [comment]
I'm wondering what's happening here (what effort is needed to get this merged in). Running into the same issue.

**Comment by frantic1048** [comment]
@moonray There is nothing special is blocking #706, just regular tests and related essential works.

Why I haven't update PR for a while: Because of many other tasks other than this `gql-gen` issue needed to be done in my work last week, and this issue was not the only one blocking my task about `gql-gen`(actually I'm migrating from `apollo-graphql` to `graphql-code-generator`), debugging further unknown issue is consuming  too much time, I left the PR here and focused on other tasks. Maybe this weekend I'll have a chance to deal with it. However, if someone is very urgent on this issue and could continue the PR is also appreciated.

**Comment by dotansimha** [comment]
Fixed in: https://github.com/dotansimha/graphql-code-generator/pull/855

**Comment by dotansimha** [comment]
Fixed in 0.14 🎉
